### ISSUE TITLE
intero has been archived

### ISSUE THREAD

**Issue by tarsius** [jojojames/smart-jump#27]
@chrisdone has archived [intero](https://github.com/commercialhaskell/intero). You probably should stop supporting it here.

Also see melpa/melpa#7225.

**Comment by tarsius** [comment]
Off topic: What's going on with Haskell support in Emacs?! It feels like huge efforts are being made and then it is all abandoned again and some other effort becomes the new raising star only to die a year or two later. I have watched that play out many times now in my role as maintainer of the Emacsmirror and am just curious. I mean in this round of looking at abandoned packages there was also [ghc(-mod)](https://github.com/DanielG/ghc-mod). Two out of ~30 archived packages are major Haskell development tools. :scream:

**Comment by jojojames** [comment]
Done, thanks for the notice.

> It feels like huge efforts are being made and then it is all abandoned again and some other effort becomes the new raising star only to die a year or two later. I have watched that play out many times now in my role as maintainer of the Emacsmirror and am just curious.

Yeah, seems like https://github.com/jyp/dante might still be up to date. With the recent lsp-esque servers cropping up and developer mindshare going to that, I'm expecting more and more language-specific support to move to or get abandoned. I'm not sure if you have thoughts on that but would be interested to hear.

**Comment by tarsius** [comment]
I'm just curious without having much to contribute to that conversation myself.

**Comment by chrisdone** [comment]
> Off topic: What's going on with Haskell support in Emacs?! It feels like huge efforts are being made and then it is all abandoned again and some other effort becomes the new raising star only to die a year or two later. I have watched that play out many times now in my role as maintainer of the Emacsmirror and am just curious. I mean in this round of looking at abandoned packages there was also [ghc(-mod)](https://github.com/DanielG/ghc-mod). Two out of ~30 archived packages are major Haskell development tools. 😱

Here’s a list: https://github.com/haskell-ide-backends-archive

Intero author here. Tooling and stability is a secondary concern for the GHC authors, they release breaking changes on a regular basis, and the current release cycle is every 6 months. Everything folds outwards from there, the standard build tool until recently just “figured out” what versions of things to combine to build your project. With that example set, maintainers of libraries follow suit. If I want to use a modern GHC, I can’t build most of my projects that are just a couple years old.

For Haskell IDE tooling, it’s impossible to keep up with it and keep your codebase tidy and stable across platforms. I gave up on Intero because a new GHC version came out and I tried for an hour to rewrite everything that broke and realised I had other more fun projects to work on. The churn is real.

I’m sure Haskell-language-server will be forked or rewritten, especially once the current caretakers leave. It only happened because a company with deep pockets funded its initial development.
